spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1021,6 +1012,52 @@ else if (channelTransacted) {
 		return receiveMessage;
 	}
 
+	public Delivery getDeliveryAndCancelConsumer(Channel channel, CompletableFuture<Delivery> future,
+			DefaultConsumer consumer, long timeoutMillis) {
+		Delivery delivery = null;
+		Throwable exception = null;
+		try {
+			if (timeoutMillis < 0) {
+				delivery = future.get();
+			}
+			else {
+				delivery = future.get(timeoutMillis, TimeUnit.MILLISECONDS);
+			}
+		}
+		catch (ExecutionException e) {
+			this.logger.error("Failed to receive message", e.getCause());
+			exception = e.getCause();
+		}
+		catch (InterruptedException e) {
+			Thread.currentThread().interrupt();
+		}
+		catch (TimeoutException e) {
+			// no result in time
+		}
+		try {
+			if (exception == null || !(exception instanceof ConsumerCancelledException)) {
+				channel.basicCancel(consumer.getConsumerTag());
+			}
+		}
+		catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("Failed to cancel consumer", e);
```

Comment:
Minor. 
We don't need `logger.isDebugEnabled()` if our log string is constant.
Although that might be better to include to the log at least `consumerTag`...

OTOH is that really intentional to `DEBUG` error problem?